backend/src/v2/metadata/client.go

```diff
@@ -846,14 +855,25 @@ func (c *Client) GetExecutionsInDAG(ctx context.Context, dag *DAG, pipeline *Pip
 		glog.V(4).Infof("execs: %v", execs)
 		for _, e := range execs {
 			execution := &Execution{execution: e}
-			taskName := execution.TaskName()
+			taskName := UpdateTaskNameWithDagID(execution.TaskName(), parentDAGID)
+			glog.V(4).Infof("taskName: %s", taskName)
+			glog.V(4).Infof("execution: %s", execution)
 			if taskName == "" {
 				if e.GetCustomProperties()[keyParentDagID] != nil {
 					return nil, fmt.Errorf("empty task name for execution ID: %v", execution.GetID())
 				}
 				// When retrieving executions without the parentDAGFilter, the rootDAG execution is supplied but does not have an associated TaskName nor is the parentDagID set, therefore we won't include it in the executionsMap.
 				continue
 			}
+			// Handle for parallelFor subdags & their tasks that consume the values from the iterator.
+			if e.GetCustomProperties()[keyIterationIndex] != nil {
+				taskName = GetParallelForTaskName(taskName, e.GetCustomProperties()[keyIterationIndex].GetIntValue())
+
+			} else if dag.Execution.GetExecution().GetCustomProperties()[keyIterationIndex] != nil {
+				// Handle for tasks within a parallelFor subdag that do not consume the values from the iterator as input but rather the output of a task that does.
```

Comment:
Appreciate the comment. The level of nesting and complexity here is **wild** (and unavoidable).